tests/FSharpPlus.Tests/SeqT.fs

```diff
@@ -0,0 +1,171 @@
+        let y3 = SeqT.run x3 |> extract |> toList
+        CollectionAssert.AreEqual (y3, [("0", 0, 0); ("1", 10, 1); ("2", 20, 2)])
+
+    [<Test>]
+    let zipLists2 () =
+        let infinite: SeqT<Async<_>, _> = SeqT.unfoldM (fun x -> monad { return if x = 13 then failwith "Unlucky number" else (Some (x, x + 1) ) }) 0
+        let finite:   SeqT<Async<_>, _> = SeqT.unfoldM (fun x -> monad { return if x = 7 then None else (Some (x, x + 1) ) }) 0
+        let x1 = zip (map string infinite) finite
+        let y1 = SeqT.run x1 |> extract |> toList
+        CollectionAssert.AreEqual (y1, [("0", 0); ("1", 1); ("2", 2); ("3", 3); ("4", 4); ("5", 5); ("6", 6)])
+        
+        let x2 = lift2 tuple2 (map string finite) (take 2 infinite)
+        let y2 = SeqT.run x2 |> extract |> toList
+        CollectionAssert.AreEqual (y2, [("0", 0); ("0", 1); ("1", 0); ("1", 1); ("2", 0); ("2", 1); ("3", 0); ("3", 1); ("4", 0); ("4", 1); ("5", 0); ("5", 1); ("6", 0); ("6", 1)])
+        
+        let x3 = SeqT.zip3 (map string infinite) (map ((*) 10) infinite) (take 3 finite)
+        let y3 = SeqT.run x3 |> extract |> toList
+        CollectionAssert.AreEqual (y3, [("0", 0, 0); ("1", 10, 1); ("2", 20, 2)])
+
+    // Compile tests
+    let binds () =
+        let res1 = SeqT [|seq [1..4] |] >>= fun x -> SeqT [|seq [x * 2] |]
+        let res2 = SeqT.hoist [1..4] >>= (fun x -> SeqT (Task.FromResult (seq [x * 2])))
+        let res3 = SeqT (ResizeArray [seq [1..4] ]) >>= (fun x -> SeqT (ResizeArray [seq [x * 2] ]))
+        let res4 = SeqT (lazy (seq [1..4])) >>= (fun x -> SeqT (lazy (seq [x * 2])))
+        let res5 = SeqT (seq [seq [1..4] ]) >>= (fun x -> SeqT (seq [seq [x * 2] ]))
+        ()
+        
+    let bind_for_identity () =
+        let res = SeqT (Identity (seq [1..4])) >>= fun x -> SeqT (Identity (seq [x * 2]))
+        ()    
+        
+    let computation_expressions () =
+        let oneTwoThree : SeqT<_, _> = monad.plus { 
+            do! lift <| Async.Sleep 10
+            yield 1
+            do! lift <| Async.Sleep 50
+            yield 2 
+            yield 3}
+        ()
+
+    let applicatives () =
+        let x = (+) <!> SeqT None <*> SeqT.ofSeq [1;2;3;4]
+        let y = (+) <!> SeqT (async.Return (seq [1])) <*> SeqT (async.Return (seq [2]))
+        let z = (+) <!> SeqT (Task.FromResult (seq [1])) <*> SeqT (Task.FromResult (seq [2]))
+        ()
+
+    let monadTransOps () =
+        let fn : SeqT<Reader<int, bool>, int> = 
+            monad.plus {
+                let! x1 = ask
+                let! x2 = 
+                    if x1 > 0 then result 1
+                    else empty
+                return x1 + x2
+            }
+        
+        let x = (fn |> SeqT.run |> Reader.run) 10 |> Seq.toList
+        areEqual [11] x
+        let y = (fn |> SeqT.run |> Reader.run) -1 |> Seq.toList
+        areEqual [] y
+
+
+module ComputationExpressions =
+
+    type __ = bool
+    
+    type Task = Task<__>
+    type Async = Async<__>
```

Comment:
I haven't seen the other code in depth, so maybe this is a standard pattern, but considering you only use `__` here, is there a difference to not just using `Task<bool>`? Seems clearer that way.

Reply:
Yes. That's unreadable without context.
The context is, I have another PR where I use some kind of consistent Higher Kind encodings, there `__` represents an unapplied type parameter. There `__` is a singleton type, but here it turned out that `bool` works better as it provides some useful information to the Enumerable abstraction. Then I found out that if I consider `__` is `bool`  I have the same notation as in that PR.
I still have to experiment with other monads.

To give you an example, it would read better ``SeqT<Async, 'T>` than `SeqT<Async<bool>, 'T>` because bool is implementation details.